### PR TITLE
Intermediate provider selection screen added

### PR DESCRIPTION
**Pull request by NikolaiYurchenko**
### Type

Feature: Intermediate provider selection screen for crypto Buy

### Context

MoonPay widget integration for Buy screen

### Parts of the app affected / Test plan

Buy screen -> stack navigator

### CODE REVIEW

File: flow-typed/npm/jest_v26.x.x.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-// flow-typed signature: e60c7806ec0ddaf4588f438843ef37bd
```
Comment (gre) [inline flow-typed/npm/jest_v26.x.x.js:1]: this PR shouldn't change the flow-typed as it's only deliverying UI changes. if there is a reason to upgrade flow-typed please do in a separate PR.

**Review by gre — CHANGES_REQUESTED** [review]

File: flow-typed/npm/jest_v26.x.x.js (hunk 1 above)
Comment (NikolaiYurchenko) [reply flow-typed/npm/jest_v26.x.x.js:1]: thanks! asked about this in chat , but had no response. will revert this

**Comment by IAmMorrow** [comment]
Hello boosty team ! I can't build locally this PR and it's probably because you guys updated the podfile. Could you please revert it ? React-native is a very fragile thing :)

**Comment by NikolaiYurchenko** [comment]
> Hello boosty team ! I can't build locally this PR and it's probably because you guys updated the podfile. Could you please revert it ? React-native is a very fragile thing :)

Done

**Review by gre — APPROVED** [review]

File: src/locales/index.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
+exports.ar = { common: require("./ar/common.json"), };
```
Comment (LFBarreto) [inline src/locales/index.js:1]: could you remove the 'ar'? i think its outside of the scope of this feature :pray:

File: src/screens/Exchange/Buy.js [hunk 3]
```diff
@@ -1,25 +1,50 @@
...
+        platform: "moonpay",
+      },
+    });
+  }, []);
+
+  const navigateToCoinify = useCallback(() => {
+    navigation.navigate(ScreenName.Coinify);
+  });
+
+  const onContinue = () => {
```
Comment (LFBarreto) [inline src/screens/Exchange/Buy.js:41]: probably useCallback this too

File: src/screens/Exchange/Buy.js [hunk 4]
```diff
@@ -1,25 +1,50 @@
...
+      screen: ScreenName.PlatformCatalog,
+      params: {
+        platform: "moonpay",
+      },
+    });
+  }, []);
+
+  const navigateToCoinify = useCallback(() => {
+    navigation.navigate(ScreenName.Coinify);
+  });
```
Comment (LFBarreto) [inline src/screens/Exchange/Buy.js:39]: you need to add navigation as deps

File: src/screens/Exchange/Buy.js [hunk 5]
```diff
@@ -1,25 +1,50 @@
...
+  const [provider, setProvider] = useState<Provider>(null);
+
+  const navigateToMoonPay = useCallback(() => {
+    navigation.navigate(NavigatorName.Platform, {
+      screen: ScreenName.PlatformCatalog,
+      params: {
+        platform: "moonpay",
+      },
+    });
+  }, []);
```
Comment (LFBarreto) [inline src/screens/Exchange/Buy.js:35]: you need to add navigation as deps

File: src/screens/Exchange/Coinify.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        <LText style={styles.description} color="smoke">
+          {t("exchange.buy.description")}
+        </LText>
+        <View style={styles.buttonContainer}>
+          <Button
+            containerStyle={styles.button}
+            event="ExchangeStartBuyFlow"
+            type="primary"
+            title={t("exchange.buy.CTAButton")}
+            onPress={() =>
```
Comment (LFBarreto) [inline src/screens/Exchange/Coinify.js:53]: preferably put this in a useCallback

**Review by LFBarreto — CHANGES_REQUESTED** [review]
some minor issues to fix :+1:

**Review by LFBarreto — APPROVED** [review]

**Comment by valpinkman** [comment]
@NikolaiYurchenko please rebase this PR with develop

**Comment by NikolaiYurchenko** [comment]
@IAmMorrow please re-check

**Comment by NikolaiYurchenko** [comment]
merged in fork PR
